DuckDuckGo/SettingsViewModel.swift

```diff
@@ -794,8 +796,18 @@ extension SettingsViewModel {
             self.state.subscription.entitlements = currentEntitlements
             self.state.subscription.subscriptionFeatures = await subscriptionManager.currentSubscriptionFeatures()
 
-        case .failure:
-            break
+        case .failure(let subscriptionServiceError):
+            if case let .apiError(apiError) = subscriptionServiceError,
+               case let .serverError(statusCode, error) = apiError {
+                if statusCode == 400 && error == "No subscription found" {
+                    state.subscription.hasSubscription = false
+                    state.subscription.hasActiveSubscription = false
+                    state.subscription.entitlements = []
+                    state.subscription.platform = .unknown
```

Comment:
Reseting the subscription state on receiving explicitly `40 "No subscription found"` error.